### ISSUE TITLE
Upgrade cancelled by “too many open files” failures

### ISSUE THREAD

**Issue by michipili** [ocaml/opam#2341]
A somewhat large upgrade job hit the roof this morning, while upgrading 34 packages and reinstalling 61 packages, the job failed with

```
[ERROR] Actions cancelled because of Sys_error("/opt/opam/log/log-4955-e3ca28.env: Too many open files")
```

Maybe we would need to throttle a bit opam's enthusiasm! :)

**Comment by AltGr** [comment]
I remember an issue related to a leak of fds was fixed some time ago ; what version are you using ?

**Comment by AltGr** [comment]
from the changelog, the fix is in 1.2.2:
* Fixed a fd leak on solver calls (#2134)

**Comment by michipili** [comment]
I still use 1.2.1, the newest 1.2.2 still did not land on my OS, :)

```
opam --version
1.2.1
```

We probably can close this issue then.
